programs/move_loader_api/src/lib.rs

```diff
@@ -79,13 +99,31 @@ pub fn execute(
     let main_module = script.into_module();
     let module_id = main_module.self_id();
     module_cache.cache_module(main_module);
-    let txn_metadata = TransactionMetadata::default();
+    let mut txn_metadata = TransactionMetadata::default();
+    txn_metadata.sender = invoke_info.sender_address;
 
     let mut vm = TransactionExecutor::new(&module_cache, data_store, txn_metadata);
-    let result = vm.execute_function(&module_id, &"main".to_string(), arguments_to_locals(args));
+    let result = vm.execute_function(
+        &module_id,
+        &"main".to_string(),
+        arguments_to_locals(invoke_info.args),
+    );
     vm.make_write_set(vec![], result).unwrap()
 }
 
+fn keyed_accounts_to_data_store(keyed_accounts: &[KeyedAccount]) -> DataStore {
```

Comment:
This is interesting. I was thinking `[KeyedAccount]` would implement some trait such as `StateView`, so that we wouldn't need to `deserialize()`.  Otherwise, good to see that borrow when calling `apply_write_set()` - hopefully it's not calling `clone()` on the other end.

Reply:
Keeping Libra thingies in the move program itself right now.  Are you fine with leaking libra types elsewhere in Solana at this stage?

Reply:
I'd prefer not to leak libra types elsewhere.  Let's first measure TPS with a naive embedding, and then debate whether deeper integration is desirable.